service/app/controllers/notification/NotificationValidator.java

```diff
@@ -0,0 +1,92 @@
+package controllers.notification;
+
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.BaseException;
+import org.sunbird.JsonKey;
+import org.sunbird.message.IResponseMessage;
+import org.sunbird.message.ResponseCode;
+import org.sunbird.request.Request;
+
+import java.text.MessageFormat;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.collections.CollectionUtils;
+
+/**
+ * Validates send notification api request
+ */
+public class NotificationValidator {
+
+
+    public static void validateSendNotificationRequest(Request request) throws BaseException {
+        List<Map<String, Object>> sendReq = (List<Map<String, Object>>) request.getRequest().get(JsonKey.NOTIFICATIONS);
+        checkMandatoryParamsPresent(sendReq, JsonKey.NOTIFICATIONS, Arrays.asList(JsonKey.DELIVERY_TYPE, JsonKey.MODE, JsonKey.IDS));
+    }
+
+    private static void checkMandatoryParamsPresent(
```

Comment:
Why can't we map this to a POJO @manzarul ? At least for these, we can afford to keep modifying it. Why choose primitive data types and struggle. I think this can be greatly simplified that way.

Reply:
Yes we can use Pojo for direct mapping.